qcengine/programs/empirical_dispersion_resources.py

```diff
@@ -795,9 +795,85 @@
             "mp2": {"params": {"s8": 1.187, "a1": 0.944, "a2": 0.480, "rcut": 0.72, "w": 0.20}}  # Rezac:2018:4711
         },
     },
+    "d4bj": {
+        "formal": "D4(BJ)",
+        "alias": ["d4"],
+        "description": "    Grimme's -D4 (BJ-damping) Dispersion Correction with ATM",
+        "citation": "    Caldeweyher, E.; Ehlert, S.; Hansen, A.; Neugebauer, H.; Spicher, S.; Bannwarth, C.; Grimmme, S., J. Chem. Phys. 150, 154122 (2019)\n",
+        "bibtex": "Caldeweyher:2019:154122",
+        "doi": "10.1063/1.5090222150",
+        "default": collections.OrderedDict([("s8", 1.0), ("a1", 1.0), ("a2", 1.0), ("s9", 1.0)]),
+        "definitions": {
+            # D4 parameters loaded below from authoritative source below. Keep a couple for reference
+            # "b3lyp": {"params": {"s8": 2.02929367, "a1": 0.40868035, "a2": 4.53807137, "s9": 1.0}},
+            # "pbe": {"params": {"s8": 0.95948085, "a1": 0.38574991, "a2": 4.80688534, "s9": 1.0}},
+        },
+    },
 }
 
 
+def _get_d4bj_definitions() -> dict:
+    """DFTD4 provides access to damping parameters on per functional basis.
+    But we want all of them.
+
+    We let DFTD4 take care of finding and loading the parameter file,
+    but to get all parameters, we implement the logic to read those
+    parameters ourselves again.
+    """
+
+    try:
+        from dftd4.parameters import get_data_file_name, load_data_base
+    except ModuleNotFoundError:
+        return {}
+
+    def get_params(entry: dict, base: dict, defaults: list) -> dict:
+        """Retrive the parameters from the data base, make sure the default
+        values are applied correctly in the process. In case we have multiple
+        defaults search for the first of the list defined for this method."""
+
+        for default in defaults:
+            try:
+                params = base[default].copy()
+                params.update(**entry[default])
+                return params
+            except KeyError:
+                continue
+
+        raise KeyError("No entry for " + method + " in parameter data base")
+
+    try:
+        _data_base = load_data_base(get_data_file_name())
+    except FileNotFoundError:
+        return {}
+
+    try:
+        _defaults = _data_base["default"]["d4"]
+        _base = _data_base["default"]["parameter"]["d4"]
+        _parameters = _data_base["parameter"]
+    except KeyError:
+        return {}
+
+    definitions = {}
+
+    for method in _parameters:
+        try:
+            _entry = _parameters[method]["d4"]
+            params = get_params(_entry, _base, _defaults)
+
+            citation = params.pop("doi", None)
+            definitions[method] = dict(
+                params=params,
+                citation=citation,
+            )
```

Comment:
```suggestion
            )
            definitions[method]["params"].pop("damping")
            definitions[method]["params"].pop("mbd")
```

The non-bj damping schemes for dftd4 would get a separate block in the big `dashcoeff` anyway. I'm not sure how mbd maps, but I think it'd be a separate block, too. So dropping these items for now if that's ok with you.

Reply:
Why not also pop damping in the changes you made?

Reply:
Deleting `damping` from `params` lead to test failures for me, seemed like it is needed somewhere in the dispersion resources, therefore I kept it.

Reply:
Sorry, your reply here didn't pop up on my browser tab until just now, hence the mention again in the consolidation comment just now https://github.com/MolSSI/QCEngine/pull/291#issuecomment-834622132.